Source/Core/VideoCommon/VertexManagerBase.cpp

```diff
@@ -193,7 +193,40 @@ u32 VertexManagerBase::GetRemainingIndices(OpcodeDecoder::Primitive primitive) c
 {
   const u32 index_len = MAXIBUFFERSIZE - m_index_generator.GetIndexLen();
 
-  if (g_Config.backend_info.bSupportsPrimitiveRestart)
+  if (primitive >= Primitive::GX_DRAW_LINES)
+  {
+    if (g_Config.UseVSForLinePointExpand())
+    {
+      const u64 div = g_Config.backend_info.bSupportsPrimitiveRestart ? 0xCCCCCCCD : 0xAAAAAAAB;
+      const u32 base = (index_len * div) >> 34;  // index_len / (prim_restart ? 5 : 6)
```

Comment:
Is this optimisation necessary? I'd expect that if it makes a significant difference, the compiler would do it (but maybe it won't do it when the divisor is from a ternary?).

Reply:
The compiler does it normally, but the ternary makes the compiler consider it non-constant and disables the optimization.  Trying to do `prim_restart ? (index_len / 5) : (index_len / 6)` has the compiler "optimize" it by factoring out `index_len`, bringing it back to the problem above.

Reply:
Huh, it seems like _no_ compiler optimises it with the ternary for the divisor: https://godbolt.org/z/WobhcMb1M

It looks like MSVC and GCC do the right thing with `prim_restart ? (index_len / 5) : (index_len / 6)`, though - only clang behaves poorly: https://godbolt.org/z/3vdfn9sxK

Using two separate switch statements depending on `g_Config.backend_info.bSupportsPrimitiveRestart` results in no `div` instruction being generated on all compilers: https://godbolt.org/z/P37EK9Wq4

IMO it would be better to do it that way instead of manually optimising the division.

Reply:
Sounds good